### ISSUE TITLE
`callback_duration.ipynb` works but `lifecycle_states.ipynb` and `memory_usage.ipynb` fails

### ISSUE THREAD

**Issue by alanxuefei** [ros-tracing/tracetools_analysis#19]
With the following setting, `callback_duration.ipynb` works but `lifecycle_states.ipynb` and `memory_usage.ipynb `
```
    # Set up the Trace action with a session name that uniquely includes a timestamp
    ros2_tracing = Trace(
        condition=IfCondition(enable_tracing),
        session_name=f'ros_tracing_navigation_session_{timestamp}',
        events_kernel=[], 
        events_ust=['ros2:*'],  # Tracks all user-space ROS 2 events
    )
```

The errors of `lifecycle_states.ipynb` are as below:

```
data_util = Ros2DataModelUtil(handler.data)

state_intervals = data_util.get_lifecycle_node_state_intervals()
for handle, states in state_intervals.items():
    print(handle)
    print(states.to_string())

output_notebook()
psize = 450
```
```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
File /usr/local/lib/python3.10/dist-packages/pandas/core/indexes/base.py:3805, in Index.get_loc(self, key)
   3804 try:
-> 3805     return self._engine.get_loc(casted_key)
   3806 except KeyError as err:

File index.pyx:167, in pandas._libs.index.IndexEngine.get_loc()

File index.pyx:196, in pandas._libs.index.IndexEngine.get_loc()

File pandas/_libs/hashtable_class_helper.pxi:2606, in pandas._libs.hashtable.Int64HashTable.get_item()

File pandas/_libs/hashtable_class_helper.pxi:2630, in pandas._libs.hashtable.Int64HashTable.get_item()

KeyError: 0

The above exception was the direct cause of the following exception:

KeyError                                  Traceback (most recent call last)
Cell In[5], line 3
      1 data_util = Ros2DataModelUtil(handler.data)
----> 3 state_intervals = data_util.get_lifecycle_node_state_intervals()
      4 for handle, states in state_intervals.items():
      5     print(handle)

File ~/src/ag_navigation/amr_navigation/external/tracetools_analysis/tracetools_analysis/analysis/../tracetools_analysis/utils/ros2.py:540, in Ros2DataModelUtil.get_lifecycle_node_state_intervals(self)
    536 node_creation_timestamp = self.data.nodes.loc[lifecycle_node_handle, 'timestamp']
    538 # Add initial and final timestamps
    539 # Last states has an unknown end timestamp
--> 540 first_state_label = transitions.loc[0, 'start_label']
    541 last_state_label = transitions.loc[transitions.index[-1], 'goal_label']
    542 transitions.loc[-1] = ['', first_state_label, node_creation_timestamp]

File /usr/local/lib/python3.10/dist-packages/pandas/core/indexing.py:1183, in _LocationIndexer.__getitem__(self, key)
   1181     key = tuple(com.apply_if_callable(x, self.obj) for x in key)
   1182     if self._is_scalar_access(key):
-> 1183         return self.obj._get_value(*key, takeable=self._takeable)
   1184     return self._getitem_tuple(key)
   1185 else:
   1186     # we by definition only have the 0th axis

File /usr/local/lib/python3.10/dist-packages/pandas/core/frame.py:4221, in DataFrame._get_value(self, index, col, takeable)
   4215 engine = self.index._engine
   4217 if not isinstance(self.index, MultiIndex):
   4218     # CategoricalIndex: Trying to use the engine fastpath may give incorrect
   4219     #  results if our categories are integers that dont match our codes
   4220     # IntervalIndex: IntervalTree has no get_loc
-> 4221     row = self.index.get_loc(index)
   4222     return series._values[row]
   4224 # For MultiIndex going through engine effectively restricts us to
   4225 #  same-length tuples; see test_get_set_value_no_partial_indexing

File /usr/local/lib/python3.10/dist-packages/pandas/core/indexes/base.py:3812, in Index.get_loc(self, key)
   3807     if isinstance(casted_key, slice) or (
   3808         isinstance(casted_key, abc.Iterable)
   3809         and any(isinstance(x, slice) for x in casted_key)
   3810     ):
   3811         raise InvalidIndexError(key)
-> 3812     raise KeyError(key) from err
   3813 except TypeError:
   3814     # If we have a listlike key, _check_indexing_error will raise
   3815     #  InvalidIndexError. Otherwise we fall through and re-raise
   3816     #  the TypeError.
   3817     self._check_indexing_error(key)

KeyError: 0
```

**Comment by christophebedard** [comment]
Can you share the trace that results in this error? I can try reproducing this when I get some time, but having the trace would be ideal.

**Comment by christophebedard** [comment]
I think your trace might just not contain any lifecycle transitions. In that case, `get_lifecycle_node_state_intervals()` should just return an empty `dict` instead of raising an error. I've fixed this in #28.

I'll close this, but feel free to open it if it still doesn't work.
